test/behavior/atomics.zig

```diff
@@ -38,6 +38,38 @@ fn testCmpxchg() !void {
     try expect(x == 42);
 }
 
+test "@cmpxchg truncates ptr" {
```

Comment:
This test passes on C backend even though the codegen is incorrect there too.
Comparing to a compile time constant like in my repro in the original issue fixes that.

Reply:
Yeah, my initial idea on how to solve this was wrong. I'll get back to this PR in a couple days, a bit busy with the RISC-V backend.